kin-ecosystem-sdk/src/main/java/com/kin/ecosystem/util/ErrorUtil.java

```diff
@@ -0,0 +1,125 @@
+	private static final String THE_REQUESTED_ACCOUNT_COULD_NOT_BE_FOUND = "The requested account could not be found";
+	private static final String FAILED_TO_ACTIVATE_ON_THE_BLOCKCHAIN_NETWORK = "A Wallet was created locally, but failed to activate on the blockchain network";
+	private static final String ECOSYSTEM_SDK_IS_NOT_STARTED = "Operation not permitted: Ecosystem SDK is not started";
+	private static final String BAD_OR_MISSING_PARAMETERS = "Bad or missing parameters";
+
+	public static KinEcosystemException fromApiException(ApiException apiException) {
+		KinEcosystemException exception;
+		if (apiException == null) {
+			exception = getUnknownException(null);
+		} else {
+			final int apiCode = apiException.getCode();
+			switch (apiCode) {
+				case 400:
+				case 401:
+				case 404:
+				case 409:
+				case 500:
+					exception = new ServiceException(SERVICE_ERROR, THE_ECOSYSTEM_SERVER_RETURNED_AN_ERROR,
+						apiException);
+					break;
+				case 408:
+					exception = new ServiceException(TIMEOUT_ERROR, ECOSYSTEM_SDK_ENCOUNTERED_AN_UNEXPECTED_ERROR,
+						apiException);
+					break;
+				case INTERNAL_INCONSISTENCY:
+					exception = new ClientException(INTERNAL_INCONSISTENCY, THE_OPERATION_TIMED_OUT, apiException);
+					break;
+				default:
+					exception = getUnknownException(apiException);
+					break;
+			}
+		}
+		return exception;
+	}
+
+	private static KinEcosystemException getUnknownException(@Nullable Throwable throwable) {
+		return new KinEcosystemException(UNKNOWN, ECOSYSTEM_SDK_ENCOUNTERED_AN_UNEXPECTED_ERROR, throwable);
+	}
+
+	public static ApiException getTimeoutException() {
+		final String errorTitle = "Time out";
+		final String errorMsg = "order timed out";
+		final int apiCode = 408;
+		ApiException apiException = new ApiException(apiCode, errorTitle);
+		apiException.setResponseBody(new Error(errorTitle, errorMsg, TIMEOUT_ERROR));
+		return apiException;
+	}
+
+	public static BlockchainException getBlockchainException(Exception error) {
+		final BlockchainException exception;
+		if (error instanceof InsufficientKinException) {
+			exception = new BlockchainException(INSUFFICIENT_KIN,
+				YOU_DO_NOT_HAVE_ENOUGH_KIN, error);
+		} else if (error instanceof TransactionFailedException) {
+			exception = new BlockchainException(TRANSACTION_FAILED,
+				THE_TRANSACTION_OPERATION_FAILED, error);
+		} else if (error instanceof CreateAccountException) {
+			exception = new BlockchainException(ACCOUNT_CREATION_FAILED, FAILED_TO_CREATE_A_BLOCKCHAIN_WALLET_KEYPAIR,
+				error);
+		} else if (error instanceof AccountNotFoundException) {
+			exception = new BlockchainException(ACCOUNT_NOT_FOUND, THE_REQUESTED_ACCOUNT_COULD_NOT_BE_FOUND, error);
+		} else if (error instanceof AccountNotActivatedException) {
+			exception = new BlockchainException(ACCOUNT_ACTIVATION_FAILED,
+				FAILED_TO_ACTIVATE_ON_THE_BLOCKCHAIN_NETWORK, error);
+		} else {
+			exception = new BlockchainException(UNKNOWN, "", error);
```

Comment:
instead of. "" we can write "unknown exception see underlying cause for more details"